skbio/metadata/_interval.py

```diff
@@ -0,0 +1,312 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
+    '''Store the metadata of a sequence interval.
+
+    Parameters
+    ----------
+    intervals : list of tuple of ints
+        List of tuples representing start and end coordinates.
+    boundaries : list of tuple of bool
+        List of tuples, representing the openness of each interval.
+    metadata : dict
+        Dictionary of attributes storing information of the feature
+        such as `strand`, `gene_name` or `product`.
+    interval_metadata : object
+        A reference to the `IntervalMetadata` object that this
+        interval is associated to.
+    '''
+    def __init__(self, intervals=None, boundaries=None,
+                 metadata=None, interval_metadata=None):
+        iv = []
+        for interval in intervals:
+            iv.append(_polish_interval(interval))
+
+        if boundaries is not None:
+            self.boundaries = boundaries
+        else:
+            self.boundaries = []
+
+        if metadata is not None:
+            self.metadata = metadata
+        else:
+            self.metadata = {}
+
+        self._interval_metadata = interval_metadata
+
+        if intervals is not None:
+            self.intervals = iv
+        else:
+            self.intervals = []
```

Comment:
hmm, the init is not right. It should behavior like `IntervalMetadata.add`. It creates a feature and associates it to a interval metadata, which means it adds the feature into the interval metadata. but it is not:
```
im = IntervalMetadata()
f = Interval(intervals=[(9, 12), (4, 5)], metadata={'name': 'sagA', 'function': 'transport'}, interval_metadata=im)
im.query([(9,12)]  # shouldn't return []
```